client/app/bundles/course/experience-points/ExperiencePointsFilterDownload.tsx

```diff
@@ -0,0 +1,101 @@
+import { Dispatch, FC, SetStateAction } from 'react';
+import { defineMessages } from 'react-intl';
+import { Download } from '@mui/icons-material';
+import {
+  Autocomplete,
+  Grid,
+  IconButton,
+  TextField,
+  Tooltip,
+} from '@mui/material';
+import {
+  ExperiencePointsFilterData,
+  ExperiencePointsNameFilterData,
+} from 'types/course/experiencePointsRecords';
+
+import useTranslation from 'lib/hooks/useTranslation';
+
+interface Props {
+  filter: ExperiencePointsFilterData;
+  disabled: boolean;
+  selectedFilter: {
+    name: ExperiencePointsNameFilterData | null;
+  };
+  setSelectedFilter: Dispatch<
+    SetStateAction<{
+      name: ExperiencePointsNameFilterData | null;
+    }>
+  >;
+  setPageNum: Dispatch<SetStateAction<number>>;
+  onClick: () => void;
+}
+
+const translations = defineMessages({
+  filterByNameButton: {
+    id: 'course.experiencePoints.filterByNameButton',
+    defaultMessage: 'Filter by Name',
+  },
+  downloadCsvButton: {
+    id: 'course.experiencePoints.downloadCsvButton',
+    defaultMessage: 'Download CSV',
+  },
+});
+
+const ExperiencePointsFilterDownload: FC<Props> = (props) => {
+  const {
+    filter,
+    selectedFilter,
+    setSelectedFilter,
+    setPageNum,
+    onClick,
+    disabled,
+  } = props;
+
+  const { t } = useTranslation();
+
+  return (
+    <div className="flex w-full justify-between">
+      <Grid columns={{ xs: 1, md: 3 }} container>
+        <Grid item paddingBottom={1} paddingRight={1} xs={1}>
+          <Autocomplete
+            key="experience-points-name-selector"
```

Comment:
Do we need the key here?

Reply:
hmm this is weird. last time I removed the key, the linter for this file produces an error that key is needed. Now seems like not anymore

Reply:
A “key” is a special string attribute you need to include when creating lists of elements in React.
There's no list here, so shouldn't be needed.
Then if its not needed then why don't we remove it before resolving?